Parsers/ASimDns/Product parsers/vimDnsMicrosoftOMS.yaml

```diff
@@ -0,0 +1,127 @@
+Parser:
+  Title: Microsoft DNS
+  Version: '0.0'
+  LastUpdated: May 10, 2021
+Product:
+  Name: MS DNS Events
+Normalization:
+  Schema: Dns
+  Version: '1.2.0'
+References:
+- Title: ASIM DNS Schema
+  Link: https://aka.ms/AzSentinelDnsDoc
+- Title: ASIM
+  Link: https://aka.ms/AzSentinelNormalization
+Description: |
+  This is a Query Parser that is used to map MS DNS (DnsEvents) to the Azure Sentinel Information Model Dns schema.  
+ParserName: vimDnsMicrosoftOMS
+ParserParams:
+  - Name: starttime
+    Type: datetime
+    Default: datetime(null)
+  - Name: endtime
+    Type: datetime
+    Default: datetime(null)
+  - Name: srcipaddr
+    Type: string
+    Default: '*'
+  - Name: domain_has_any
+    Type: dynamic
+    Default: dynamic([])
+  - Name: responsecodename
+    Type: string
+    Default: '*'
+  - Name: dnsresponsename
+    Type: string
+    Default: '*'
+  - Name: response_has_any
+    Type: dynamic
+    Default: dynamic([])
+  - Name: eventtype
+    Type: string
+    Default: 'lookup'
+ParserQuery: |
+  let RCodeTable=datatable(ResponseCode:int,ResponseCodeName:string)[
+     0, 'NOERROR'
+   , 1, "FORMERR"
+   , 2,"SERVFAIL"
+   , 3,'NXDOMAIN'
+   , 4,'NOTIMP'
+   , 5,'REFUSED'
+   , 6,'YXDOMAIN'
+   , 7,'YXRRSET'
+   , 8,'NXRRSET'
+   , 9,'NOTAUTH'
+   , 10,'NOTZONE'
+   , 11,'DSOTYPENI'
+   , 16,'BADVERS'
+   , 16,'BADSIG'
+   , 17,'BADKEY'
+   , 18,'BADTIME'
+   , 19,'BADMODE'
+   , 20,'BADNAME'
+   , 21,'BADALG'
+   , 22,'BADTRUNC'
+   , 23,'BADCOOKIE'];
+  let DNSQuery_MS=(
+    starttime:datetime=datetime(null), endtime:datetime=datetime(null)
+    , srcipaddr:string='*'
+    , domain_has_any:dynamic=dynamic([]) 
+    , responsecodename:string='*', dnsresponsename:string='*'
+    , response_has_any:dynamic=dynamic([]) , eventtype:string='lookup'
+    ){
+    let rcodenames=RCodeTable | project ResponseCodeName;
+    DnsEvents 
+    // ******************************************************************
+      //  Pre-parsing filterring:
+        | where
+          (isnull(starttime) or TimeGenerated > starttime)
+          and (isnull(endtime) or TimeGenerated < endtime) 
+          and (srcipaddr=='*' or ClientIP==srcipaddr)
+          and (array_length(domain_has_any) ==0 or Name has_any (domain_has_any))
+          and (responsecodename=='*' or ResultCode in (rcodenames))
+          and (dnsresponsename=='*' or has_ipv4(IPAddresses,dnsresponsename) )
+          and (array_length(response_has_any) ==0 or has_any_ipv4_prefix(IPAddresses, response_has_any)  )
```

Comment:
Or, getting back to my Umbrella comment, do we have both responsename (single IPv4) and response_has_any?

Reply:
we have both due to syntax limitation. with domain I could use one syntax for both  single string and a dynamic list.
This does not work with has_any_ipv4 function (I cannot pass a string)